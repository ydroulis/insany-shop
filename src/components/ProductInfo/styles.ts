"use client"

import styled from 'styled-components'

export const Container = styled.section`
color: #41414d;
width: 100%;
display: flex;
flex-direction: column;
justify-content: space-between;

@media (min-width: 768px) {
    width: 704px;
    width: 448px;
}
`

export const Details = styled.div`
    width: 100%;
`

export const Category = styled.p`
    font-size: 1rem;
    margin-bottom: 0.75rem;
`

export const Title = styled.h1`
    font-size: 2rem;
    font-weight: 300;
    margin-bottom: 0.5rem;
`

export const Price = styled.p`
    font-size: 1.25rem;
    font-weight: 600;
    color: #46ab80;
    margin-bottom: 1rem;
    
    @media (min-width: 1440px) {
        margin-bottom: 6.25rem;
    }
`

export const Description = styled.div`
    margin-bottom: 1.25rem;

    p{
        font-size: 1rem;
        font-weight: 500;
        color: #737380;
        margin-bottom: 1.25rem;
        text-transform: uppercase;
    }

    span{
        font-size: 0.875rem;
    }

    @media (min-width: 1440px) {
        margin-bottom: 0rem;
    }
`
