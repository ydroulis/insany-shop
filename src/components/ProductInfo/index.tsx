import React from 'react';

import * as S from './styles';
import Button from '../Button';

interface ProductInfoProps {
    category: string | undefined
    name: string
    price: number
    description: string
}

const ProductInfo: React.FC<ProductInfoProps> = ({
    category,
    name,
    price,
    description
}) => {
    const formattedPrice = new Intl.NumberFormat('pt-BR', {
        style: 'currency',
        currency: 'BRL',
    }).format(price);

    return (
        <S.Container
            aria-labelledby="product-title"
            aria-describedby="product-description"
        >
            <S.Details>
                <S.Category>{category}</S.Category>
                <S.Title id="product-title">{name}</S.Title>
                <S.Price aria-label="Preço do produto">{formattedPrice}</S.Price>
                <S.Description id="product-description">
                    <p>Descrição</p>
                    <span>{description}</span>
                </S.Description>
            </S.Details>
            <Button action={() => { }}>Adicionar</Button>
        </S.Container>
    );
}

export default ProductInfo;