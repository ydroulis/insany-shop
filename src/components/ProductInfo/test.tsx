import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ProductInfo from "./";

jest.mock("../../providers/cartStoreProvider", () => ({
    useCartStore: jest.fn(() => ({
        addProductToCart: jest.fn().mockResolvedValue(undefined), // retorna uma promise
    })),
}));

describe("<ProductInfo />", () => {
    const defaultProps = {
        category: "Casa e Decoração",
        name: "Sofá 3 Lugares Retrátil",
        price: 1899.99,
        description:
            "Sofá confortável com assento retrátil e reclinável, revestimento em tecido suede e estrutura de madeira maciça.",
        image: "",
    };

    it("should render product info with correct content", () => {
        render(<ProductInfo {...defaultProps} />);

        const title = screen.getByRole("heading", { name: /Sofá 3 Lugares Retrátil/i });
        expect(title).toBeInTheDocument();

        expect(screen.getByText(/Casa e Decoração/i)).toBeInTheDocument();

        const price = screen.getByLabelText("Preço do produto");
        expect(price).toBeInTheDocument();
        expect(price).toHaveAttribute("aria-label", "Preço do produto");

        const description = screen.getByText(/Sofá confortável com assento retrátil/i);
        expect(description).toBeInTheDocument();

        const container = screen.getByRole("region");
        expect(container).toHaveAttribute("aria-labelledby", "product-title");
        expect(container).toHaveAttribute("aria-describedby", "product-description");
    });

    it("should render button with correct text and be clickable", async () => {
        const user = userEvent.setup();
        render(<ProductInfo {...defaultProps} />);

        const button = screen.getByRole("button", { name: /Adicionar/i });
        expect(button).toBeInTheDocument();

        await user.click(button);
    });

    it("should match snapshot", () => {
        const { container } = render(<ProductInfo {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });
});
