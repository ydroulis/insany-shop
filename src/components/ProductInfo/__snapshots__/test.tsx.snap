// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ProductInfo /> should match snapshot 1`] = `
<div>
  <section
    aria-describedby="product-description"
    aria-labelledby="product-title"
    class="sc-bRKDuR gSxecp"
  >
    <div
      class="sc-hvigdm cSUguT"
    >
      <p
        class="sc-fhHczv hCsJkM"
      >
        Casa e Decoração
      </p>
      <h1
        class="sc-ggWZvA bBbRPB"
        id="product-title"
      >
        Sofá 3 Lugares Retrátil
      </h1>
      <p
        aria-label="Preço do produto"
        class="sc-dTvVRJ cJcKdX"
      >
        R$ 1.899,99
      </p>
      <div
        class="sc-jwTyAe cDycJE"
        id="product-description"
      >
        <p>
          Descrição
        </p>
        <span>
          Sofá confortável com assento retrátil e reclinável, revestimento em tecido suede e estrutura de madeira maciça.
        </span>
      </div>
    </div>
    <button
      aria-label="Adicionar"
      class="sc-hjsuWn jsmGPx"
      type="button"
    >
      <svg
        aria-hidden="true"
        color="#fff"
        data-testid="icon-svg"
        fill="none"
        focusable="false"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="9"
          cy="21"
          r="1"
        />
        <circle
          cx="20"
          cy="21"
          r="1"
        />
        <path
          d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"
        />
      </svg>
      <p>
        Adicionar
      </p>
    </button>
  </section>
</div>
`;
